app/src/main/java/bacon/john/bitcoinpricewatch/bitcoin/BitcoinCurrencyManager.java

```diff
@@ -0,0 +1,122 @@
+package bacon.john.bitcoinpricewatch.bitcoin;
+
+import org.json.JSONObject;
+
+import java.util.*;
+import org.json.*;
+import java.text.*;
+import io.realm.*;
+
+import bacon.john.bitcoinpricewatch.localstorage.LocalStorageService;
+
+/**
+ * @class BitcoinCurrencyManager
+ *
+ * Retrieves information about bitcoin in various currencies
+ */
+public class BitcoinCurrencyManager {
+    public interface CurrencyListCallback {
+        public void callback(List<BitcoinCurrencyModel> currencies, Exception error);
+    }
+    public interface CurrencyCallback {
+        public void callback(BitcoinCurrencyModel currency, Exception error);
+    }
+
+    private static BitcoinCurrencyManager sharedInstance;
+
+    public static BitcoinCurrencyManager sharedInstance() {
+        if (sharedInstance == null) {
+            sharedInstance = new BitcoinCurrencyManager();
+        }
+        return sharedInstance;
+    }
+
+    public List<BitcoinCurrencyModel> getCachedCurrencies() {
+        RealmResults<BitcoinCurrencyModel> results = LocalStorageService.sharedInstance().getRealm().where(BitcoinCurrencyModel.class).findAll();
+        results.sort("code");
+        return results;
+    }
+
+    public BitcoinCurrencyModel getCachedCurrency(String currencyCode) {
+        return LocalStorageService.sharedInstance().getRealm().
+                where(BitcoinCurrencyModel.class).
+                equalTo("code", currencyCode).findFirst();
+    }
+
+    public void getRemoteCurrencyCodes(final CurrencyListCallback callback) {
+        BitcoinCurrencyService.sharedInstance().fetchCurrencyCodes(new BitcoinCurrencyService.Callback() {
+            @Override
+            public void callback(final JSONObject response, Exception error) {
+                if (error != null) {
```

Comment:
ideally there would be exponential back-off logic here instead the service was down so we don't hammer it